apps/web/src/components/activity/ExecutionDetailsStepContent.tsx

```diff
@@ -52,19 +53,21 @@ export const ExecutionDetailsStepContent = ({ step }) => {
               <ExecutionDetail
                 key={`execution-detail-${executionDetail.id}`}
                 executionDetail={executionDetail}
-                showTriggerSnippet={showTriggerSnippet}
-                onShowExecutionDetail={onShowExecutionDetail}
+                showTriggerSnippet={detailId !== executionDetail.id}
+                onShowExecutionDetail={onShowExecutionDetail(executionDetail.id, executionDetail.raw)}
                 onHideExecutionDetail={onHideExecutionDetail}
               />
             ))}
           </GridColContainer>
         </Grid.Col>
         <Grid.Col span={6}>
           <GridColContainer theme={theme}>
-            {!showTriggerSnippet && <ExecutionDetailTrigger step={step} />}
-            {showTriggerSnippet && executionDetailsRawSnippet && (
+            <When truthy={detailId.length === 0}>
```

Comment:
I find it more confusing to check that a string is empty by checking its length rather that strict checking empty string. Is there any edge case that this covers?

Reply:
I do not think so just the way I learned to check an empty string 😄

Reply:
Up to your choice then, though first thought I had was why `detailId` was an array. 😅